usecases/scheduled_execution/offloading_job.go

```diff
@@ -0,0 +1,212 @@
+package scheduled_execution
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/checkmarble/marble-backend/infra"
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/repositories"
+	"github.com/checkmarble/marble-backend/usecases/executor_factory"
+	"github.com/checkmarble/marble-backend/utils"
+	"github.com/riverqueue/river"
+	"github.com/riverqueue/river/rivertype"
+	"gocloud.dev/blob"
+)
+
+type offloadingRepository interface {
+	GetOffloadedDecisionRuleKey(orgId, decisionId, ruleId string, createdAt time.Time) string
+
+	GetOffloadingWatermark(ctx context.Context, exec repositories.Executor, orgId, table string) (*models.OffloadingWatermark, error)
+	SaveOffloadingWatermark(ctx context.Context, tx repositories.Transaction,
+		orgId, table, watermarkId string, watermarkTime time.Time) error
+
+	GetOffloadableDecisionRules(ctx context.Context, exec repositories.Executor,
+		req models.OffloadDecisionRuleRequest) (<-chan repositories.ModelResult[models.OffloadableDecisionRule], error)
+	RemoveDecisionRulePayload(ctx context.Context, tx repositories.Transaction, ids []string) error
+}
+
+func NewOffloadingPeriodicJob(orgId string, interval time.Duration) *river.PeriodicJob {
+	return river.NewPeriodicJob(
+		river.PeriodicInterval(interval),
+		func() (river.JobArgs, *river.InsertOpts) {
+			return models.OffloadingArgs{
+					OrgId: orgId,
+				}, &river.InsertOpts{
+					Queue: orgId,
+					UniqueOpts: river.UniqueOpts{
+						ByQueue:  true,
+						ByPeriod: interval,
+						ByState: []rivertype.JobState{
+							rivertype.JobStateAvailable,
+							rivertype.JobStatePending, rivertype.JobStateRunning,
+							rivertype.JobStateRetryable, rivertype.JobStateScheduled,
+						},
+					},
+				}
+		},
+		&river.PeriodicJobOpts{RunOnStart: true},
+	)
+}
+
+type OffloadingWorker struct {
+	river.WorkerDefaults[models.OffloadingArgs]
+
+	executorFactory    executor_factory.ExecutorFactory
+	transactionFactory executor_factory.TransactionFactory
+	repository         offloadingRepository
+	blobRepository     repositories.BlobRepository
+
+	config infra.OffloadingConfig
+}
+
+func NewOffloadingWorker(executorFactory executor_factory.ExecutorFactory, transactionFactory executor_factory.TransactionFactory,
+	repository offloadingRepository, blobRepository repositories.BlobRepository,
+	offloadingBucketUrl string, offloadConfig infra.OffloadingConfig,
+) *OffloadingWorker {
+	return &OffloadingWorker{
+		executorFactory:    executorFactory,
+		transactionFactory: transactionFactory,
+		repository:         repository,
+		blobRepository:     blobRepository,
+		config:             offloadConfig,
+	}
+}
+
+func (w OffloadingWorker) Work(ctx context.Context, job *river.Job[models.OffloadingArgs]) error {
+	logger := utils.LoggerFromContext(ctx)
+	exec := w.executorFactory.NewExecutor()
+	timeout := time.After(w.config.JobInterval - 10*time.Second)
```

Comment:
We cannot use river's default timeout implementation, since jobs are marked as failed when it's reached, so we need to bail out when we get there.

Reply:
We actually needed to override river's `Timeout()` method so it doesn't "rescue" the job after one minute.